lib/slack/web/faraday/response/raise_error.rb

```diff
@@ -6,6 +6,7 @@ module Response
         class RaiseError < ::Faraday::Response::Middleware
           def on_complete(env)
             raise Slack::Web::Api::Errors::TooManyRequestsError, env.response if env.status == 429
+            raise Slack::Web::Api::Errors::SlackError.new("Invalid JSON received from Slack API.", env.response) if !body.is_a?(Hash)
```

Comment:
The point here is not that we got something other than JSON, but that we got back a 500. Slack API is predictable as in with a 500 you don't get JSON. So you should specialize the error similarly to `TooManyRequestsError`, and check for the error code instead of body being a hash to avoid future surprises.